HPXMLtoOpenStudio/resources/hpxml.rb

```diff
@@ -4653,6 +4658,60 @@ def from_oga(generator)
     end
   end
 
+  class Batteries < BaseArrayElement
+    def add(**kwargs)
+      self << Battery.new(@hpxml_object, **kwargs)
+    end
+
+    def from_oga(hpxml)
+      return if hpxml.nil?
+
+      XMLHelper.get_elements(hpxml, 'Building/BuildingDetails/Systems/Batteries/Battery').each do |battery|
+        self << Battery.new(@hpxml_object, battery)
+      end
+    end
+  end
+
+  class Battery < BaseElement
+    ATTRS = [:id, :type, :location,
+             :lifetime_model, :capacity, :voltage]
+    attr_accessor(*ATTRS)
+
+    def delete
+      @hpxml_object.batteries.delete(self)
+    end
+
+    def check_for_errors
+      errors = []
+      return errors
+    end
+
+    def to_oga(doc)
+      return if nil?
+
+      batteries = XMLHelper.create_elements_as_needed(doc, ['HPXML', 'Building', 'BuildingDetails', 'Systems', 'Batteries'])
+      battery = XMLHelper.add_element(batteries, 'Battery')
+      sys_id = XMLHelper.add_element(battery, 'SystemIdentifier')
+      XMLHelper.add_attribute(sys_id, 'id', @id)
+      XMLHelper.add_element(battery, 'Location', @location, :string, @location_isdefaulted) unless @location.nil?
+      XMLHelper.add_element(battery, 'BatteryType', @type, :string) unless @type.nil?
+      XMLHelper.add_element(battery, 'NominalCapacity', @capacity, :float, @capacity_isdefaulted) unless @capacity.nil?
+      XMLHelper.add_element(battery, 'NominalVoltage', @voltage, :float, @voltage_isdefaulted) unless @voltage.nil?
```

Comment:
Need to rename these variables from `capacity` and `voltage` to `nominal_capacity` and `nominal_voltage`, so it's clear what they are. (Though note that we may ultimately change to `usable_capacity` or other HPXML elements...)